azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py

```diff
@@ -39,13 +68,10 @@ def __init__(self, auth_provider):
         self._mqtt_provider = None
         self.on_transport_connected = None
         self.on_transport_disconnected = None
-        self.on_event_sent = None
-        self._event_queue = queue.LifoQueue()
-        self._event_callback_map = {}
+        self._action_queue = queue.Queue()
+        self._callback_map = {}
```

Comment:
If i understand correctly, _action_queue is for "pending" actions (ie not on the wire yet) and _callback_map is to store the callbacks of actions "in flight" (ie the packet has been sent and the ACK hasn't been received). I wonder:
- if the naming could convey this idea (provided that my understanding is correct)
- (less important) if we need one additional "action processor" data structure that groups together the pending action queue, the callback map, and the methods that act on those (_add_to_queue, _do_single_action and _do_actions_in_queue). would it make this easier to test?

Reply:
Awesome!  I totally agree on both points, but I want to refactor the "action processor" in a separate commit.

---
In reply to: [266185036](https://github.com/Azure/azure-iot-sdk-python-preview/pull/63#discussion_r266185036) [](ancestors = 266185036)